src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -879,23 +852,36 @@ def __export_keyvalues(fetched_items, format_, separator, prefix=None):
                 continue
 
             key_segments = key.split(separator)
-            if key_segments[0].isdigit():
-                __export_keyvalue(key_segments,
-                                  kv.value, exported_list, key)
-
-            else:
-                __export_keyvalue(key_segments,
-                                  kv.value, exported_dict, key)
-
-        if exported_dict and exported_list:
-            logger.error("Can not export to a valid file! Some keys have been dropped. %s", json.dumps(
-                exported_dict, indent=2, ensure_ascii=False))
+            __export_keyvalue(key_segments, kv.value, exported_dict)
 
-        return __compact_key_values(exported_dict if not exported_list else exported_list)
+        return __try_convert_to_arrays(exported_dict)
     except Exception as exception:
         raise CLIError("Fail to export key-values." + str(exception))
 
 
+def __try_convert_to_arrays(constructed_data, is_root=True):
+    if not (isinstance(constructed_data, dict) and len(constructed_data) > 0):
+        return constructed_data
+
+    # Object cannot be an array if not all keys are numeric
+    if False not in (key.isdigit() for key in constructed_data):
+        is_array = True
+        sorted_data_keys = sorted(int(key) for key in constructed_data)
```

Comment:
> sorted(int(key) for key in constructed_data) [](http://example.com/codeflow?start=26&length=45)

Wouldnt constructed_data already be sorted?

Reply:
Python dictionary keys are not sorted. They remain in the order in which they are added.

Reply:
Agreed, I was thinking more about how backend always returns data sorted by key name. So even though dictionaries are not sorted, the data in constructed_data will be sorted since its coming from our backend. But we shouldn't rely on backend sorting and its okay to sort it here just to be sure.

Reply:
Got it!  Thanks for pointing this out. I did not know that the backend returns data sorted by key name. This is not the case with the Portal so I guess I was working under the same assumption here to be safe. Thanks!